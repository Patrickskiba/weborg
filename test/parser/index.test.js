const parse = require('../../src/parser')

const text = 
`* This is a test
I think this is pretty cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
* TODO level 1
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
* yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
* level 1 headline
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool
** yea this is super cool
super super super cool`

describe('e2e parser tests', () => {
    it('reads a series of lines and creates an ast', () => {
        const ast = parse(text)
        expect(ast.length).toEqual(4)
        expect(ast[0].children.length).toEqual(8)
        expect(ast).toMatchSnapshot()
    }) 
})
