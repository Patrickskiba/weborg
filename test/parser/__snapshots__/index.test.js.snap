// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e parser tests reads a series of lines and creates an ast 1`] = `
Array [
  Object {
    "State": undefined,
    "children": Array [
      Object {
        "content": Array [
          Object {
            "text": "I think this is pretty cool",
            "type": "text",
          },
        ],
        "type": "section",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
    ],
    "content": Array [
      Object {
        "text": "This is a test",
        "type": "text",
      },
    ],
    "level": 1,
    "priority": undefined,
    "tags": undefined,
    "type": "headline",
  },
  Object {
    "State": "TODO",
    "children": Array [
      Object {
        "content": Array [
          Object {
            "text": "super super super cool",
            "type": "text",
          },
        ],
        "type": "section",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
    ],
    "content": Array [
      Object {
        "text": "level 1",
        "type": "text",
      },
    ],
    "level": 1,
    "priority": undefined,
    "tags": undefined,
    "type": "headline",
  },
  Object {
    "State": undefined,
    "children": Array [
      Object {
        "content": Array [
          Object {
            "text": "super super super cool",
            "type": "text",
          },
        ],
        "type": "section",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
    ],
    "content": Array [
      Object {
        "text": "yea this is super cool",
        "type": "text",
      },
    ],
    "level": 1,
    "priority": undefined,
    "tags": undefined,
    "type": "headline",
  },
  Object {
    "State": undefined,
    "children": Array [
      Object {
        "content": Array [
          Object {
            "text": "super super super cool",
            "type": "text",
          },
        ],
        "type": "section",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
      Object {
        "State": undefined,
        "children": Array [
          Object {
            "content": Array [
              Object {
                "text": "super super super cool",
                "type": "text",
              },
            ],
            "type": "section",
          },
        ],
        "content": Array [
          Object {
            "text": "yea this is super cool",
            "type": "text",
          },
        ],
        "level": 2,
        "priority": undefined,
        "tags": undefined,
        "type": "headline",
      },
    ],
    "content": Array [
      Object {
        "text": "level 1 headline",
        "type": "text",
      },
    ],
    "level": 1,
    "priority": undefined,
    "tags": undefined,
    "type": "headline",
  },
]
`;
